Fixed breaking didRefreshRegistrationToken problem

BenSower opened the pull request:
When building my flutter app, I ran into this problem:
https://stackoverflow.com/questions/44116626/fira-appdelegate-messagingdidrefreshregistrationtoken-unrecognized-selecto
(not my post, but exactly the same problem)

And wrote this minimalistic fix. Since I do not think I am proficient enough concerning the whole plugin and FCM, I did not implement an actual body for the new function. Is there an expected behaviour, which should be implemented?

Take care!

BenSower commented:
Trying to fix cla by commenting.

mit-mit commented:
Hi @BenSower we fixed the Travis issue in https://github.com/flutter/plugins/pull/220, so if you merge with master hopefully it will pass now.

mit-mit commented:
Travis passing now :-)

goderbauer commented:
I don't think the changes proposed in this PR are enough to adopt to the underlying changes in the Firebase SDK. For reference, see https://github.com/firebase/quickstart-ios/commit/d1ee0c62e522313c25f4af903ea0da3fb8d4d6cc and later changes to learn how the Firebase Quickstart Examples got updated. We will have to do similar changes to this plugin.

/cc @kroikie, who might be able to help with this.

kroikie commented:
Hi @BenSower thanks for the PR, @goderbauer is correct there would be a bit more required for this change.

- Implement the didRefreshRegistrationToken callback. This should do what `tokenRefreshNotification` currently does. There is no longer any need for the observer so it can be removed.
- Remove `connectToFcm`. All FCM messages both notification-messages and data-messages are now sent via APNs so there is no need to 'connect' unless you are sending upstream messages which this SDK does not yet support.
- We are still using iOS 9 handlers for incoming messages and should add the iOS 10 handlers but that can/should be done in a separate PR.

BenSower commented:
Hi @kroikie, I

- moved the logic from 'tokenRefreshNotification' to 'didRefreshRegistrationToken',
- removed 'tokenRefreshNotification' as well as the observer registration
- removed the 'connectToFcm' method and all related calls

what do you think?

BenSower commented:
@collinjackson I updated the changelog and the pubspec. Just a general question: 
What is the best practice concerning the author(s) field? Do I have to add myself to the authors field, since it's not only `Flutter Team <flutter-dev@googlegroups.com>` anymore?

mit-mit commented:
@BenSower do you mean the authors field in `pubspec.yaml`? I believe we have been using that for only the main/original author, but please do add yourself to the `AUTHORS file`: https://github.com/flutter/plugins/blob/master/AUTHORS

BenSower commented:
@mit-mit Okay, I added myself to the Authors file.

mit-mit commented:
@kroikie this one looks ready to go. Can you give it a last look, please?

BenSower commented:
@kroikie @mit-mit @collinjackson : Heyho, not trying to be rude, just wondering: Might there be another reviewer, so this pr doesn't get too old? :-)

goderbauer commented on packages/firebase_messaging/ios/Classes/FirebaseMessagingPlugin.m:
```diff
@@ -180,4 +150,10 @@ - (void)application:(UIApplication *)application
   [_channel invokeMethod:@"onIosSettingsRegistered" arguments:settingsDictionary];
 }
 
+// this method replaces the deprecated 'tokenRefreshNotification' method
```
I'd remove this comment. That's not really valuable to anybody looking at the code in the future.

BenSower replied:
Yup, good call.

goderbauer commented on packages/firebase_messaging/ios/Classes/FirebaseMessagingPlugin.m:
```diff
@@ -139,7 +110,6 @@ - (void)applicationDidEnterBackground:(UIApplication *)application {
 
 - (void)applicationDidBecomeActive:(UIApplication *)application {
```
(this comment is about the line four lijnes above this one , which github doesn't allow me to comment on).

@kroikie Do we still have to disconnect manually now that Firebase is managing the connection?

collinjackson replied:
good catch, I don't think we need to disconnect manually

BenSower replied:
Ok, I removed the disconnect. While being at it, I also updated the Changelog, since there were some external changes to it by now.

BenSower replied:
@goderbauer @collinjackson any more suggestions?

kroikie replied:
Correct we don't have to disconnect manually anymore.

kroikie commented:
Sorry for the delay, this looks good to me.

BenSower commented:
Perfect, thank you for your time and have a nice weekend!

goderbauer commented:
This is live now: https://pub.dartlang.org/packages/firebase_messaging
